c/8queens/main.c

```diff
@@ -0,0 +1,150 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <inttypes.h>
+//#include "Source File/main.h"
+
+#include <stdint.h>
+#define N (8)
+#define MAX_QUEENS (8)
+#define QUEEN_MARK (-1)
+
+typedef struct {
+    int32_t x;
+    int32_t y;
+} QueenPosition;
+
+void print_board(int32_t board[N][N]);
+void calculate_threats(int32_t board[N][N], QueenPosition queens[MAX_QUEENS], size_t queen_count);
+
+
+void print_board(int32_t board[N][N])
+{
+    for (size_t row = 0; row < N; row++)
+    {
+//        char first_line [N*2];
+//        memset(first_line, '-', sizeof(first_line));
+//        printf("%s\n", first_line);
+        for (size_t i = 0; i < N; i++)
+        {
+            printf("----");
+        }
+        printf("\n");
+        for (size_t col = 0; col < N; col++)
+        {
+            char print_ch;
+            if (board[row][col] == QUEEN_MARK)
+                print_ch = 'Q';
+            else
+                print_ch = board[row][col] + '0';
+            printf("| %c ", print_ch);
+        }
+        printf("|\n");
+    }
+}
+
+
+void calculate_threats(int32_t board[N][N], QueenPosition queens[MAX_QUEENS], size_t queen_count)
+{
+
+    int y = (queens[0].y);
+    int32_t x = board[1][(int)queens[0].y];
+    printf("%d, %d", x, y);
+
+    for (size_t queen = 0; queen < queen_count; queen++)
+    {
+        // Adding 1 to all attacked rows below queen
+        for (size_t row = (queens[queen].x) + 1; row < N; row++)
+        {
+            if (board[row][queens[queen].y] != -1)
```

Comment:
Reverse `if` and `else`. Then you can remove the `else`.
In general, `if`s are for edge cases, so that the "happy path" can be straightforward.